### PR TITLE
Add parser for telegram username

### PR DESCRIPTION
**Pull request by cshao02**
For #56 

Enabled parsing for (T/) tag instead of (t/) tag as (t/) is still reserved for optional tags for now
- Added telegram username field to Person class
- Telegram username field can be null, which represents that user did not input a telegram handle for that contact (perhaps can move towards using an Optional class to better represent the value where it could possibly be null)

Added backend logic of storing telegram username to model and storage, as well as retrieving it from json on startup

Added telegram username to GUI beside the name of the contact

Things to work on 
- Add test cases
- Check for bugs
- Potentially delete (t/) tags feature to allow telegram username to use (t/) tag instead of (T/)
- Handle current failing test cases

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 1]
```diff
@@ -31,21 +33,25 @@ public class AddCommandParser implements Parser<AddCommand> {
      */
     public AddCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG,
+                        PREFIX_TELEGRAM);
```
Comment (shardhrv) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:37]: Nit: could add multiple params to the new line

**Review by brianchoon — APPROVED** [review]
Do fix the errors in the checks in your next PR!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -80,6 +81,26 @@ public static Address parseAddress(String address) throws ParseException {
...
 
+    /**
+     * Parses the given {@code String telegramUsername} and returns a {@code TelegramUsername} object.
+     * Leading and trailing spaces will be trimmed.
+     *
+     * @param telegramUsername The username string to be parsed. Can be {@code null}.
+     * @return A {@code TelegramUsername} object with the trimmed username.
+     * @throws ParseException if the provided {@code telegramUsername} is invalid according to the
+     *         {@code TelegramUsername#isValidUsername(String)} method.
+     */
```
Comment (shardhrv) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:92]: Well documented!

**Review by shardhrv — APPROVED** [review]
LGTM! just need to address checkstyle

**Pull request merged**